accepted/2023/owner-author-pmui.md

```diff
@@ -0,0 +1,78 @@
+# ***Display Trusted Owners in PM UI***
+<!-- Replace `Title` with an appropriate title for your design -->
+
+- Donnie Goodson ([donnie-msft](https://github.com/donnie-msft))
+- GitHub Issue https://github.com/NuGet/Home/issues/12501
+
+## Summary
+
+When available, highlight in the Packages Manager UI (PM UI) that the owner is trusted by the package source with a NuGet package icon. 
+In the case of `nuget.org`, also provide a hyperlink to the owner's profile page (ie, `https://www.nuget.org/profiles/<owner name>`).
+A Trusted Owner is defined as an owner that's indicated by a package source's [Search API](https://learn.microsoft.com/nuget/api/search-query-service-resource#search-result), as opposed to the field in `.nuspec`.
+
+## Motivation 
+
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+
+Customers need a clear understanding of who created a NuGet package in order to trust that package.
+The term "owner" has become overloaded as NuGet has evolved, which has caused confusion. 
+The "author" metadata is also misleading as it's a free-form text field which doesn't imply any credentials.
+
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+
+![Owners from NuGet.org are shown beside each package ID with a hyperlink and a NuGet Package Icon](../../meta/resources/OwnerAuthor/ownerNuGetOrgLinks.png)
+
+### Technical explanation
+
+<!-- Explain the proposal in sufficient detail with implementation details, interaction models, and clarification of corner cases. -->
+
+Currently, `Owners` JSON parsed from the Search API is immediately converted into a comma-delimited string of owners.
+To avoid breaking any clients depending on this, `Owners` will continue being a pre-processed string with comma-delimited owners.
+A new property, `OwnersEnumerable` (naming TBD) will be added to [IPackageSearchMetadata](https://github.com/NuGet/NuGet.Client/blob/dc2fadacd311646754ce6b062489b403338972aa/src/NuGet.Core/NuGet.Protocol/Model/IPackageSearchMetadata.cs) and [PackageSearchMetadataContextInfo](https://github.com/NuGet/NuGet.Client/blob/dc2fadacd311646754ce6b062489b403338972aa/src/NuGet.Clients/NuGet.VisualStudio.Internal.Contracts/ContextInfos/PackageSearchMetadataContextInfo.cs).
+The enumerable owners will allow displaying custom UI for each owner.
```

Comment:
What happens if there are many, many owners? Or the cumulative comma separated list is very long?

```kusto
NiPackageOwners
| extend ['Comma\'d'] = array_strcat(Owners, ", ")
| summarize
    (['Max comma separated len'], ['Example ID 1']) = arg_max(strlen(["Comma'd"]), Id),
    (['Max owner count'], ['Example ID 2']) = arg_max(array_length(Owners), Id)
```

|Max comma separated len|Example ID 1|Max owner count|Example ID 2|
|---|---|---|---|
|192|KenticoCommunity.ConnectionStringModule|12|Cake|

Reply:
Today, the UI truncates a single line of text at the width of the packages list. The reasoning is that the packages list doesn't intend to show the full context of metadata, but instead provides a "metadata preview" to give customers a little more than "package ID" when searching. Rich details are always available in the details pane. 

This maximizes real estate in the packages list. Alternate designs may come up as we rethink the PM UI layout and architecture.

I believe I'll propose to keep that philosophy even with owners and profile hyperlinks. It's true that some owners may not be visible (depending on the actual width of the packages list), but customers can:
1. View the tooltip to find the complete owner list & package description.
2. Select the package to find a complete Owners listing in the details pane.